### ISSUE TITLE
Train problem

### ISSUE THREAD

**Issue by yzliu90** [CSTR-Edinburgh/merlin#49]
Hi:
Do you know the reason about the trouble "epoch 1, validation error nan, train error nan  time spent 31758.06"

**Comment by zhizhengwu** [comment]
It looks like something is wrong with your training data.
What is your setting? Are you using GPU?

**Comment by yzliu90** [comment]
I was using CPU

**Comment by yzliu90** [comment]
I have checked my data, the data is ok.

**Comment by ronanki** [comment]
Use below script to check for nan in your training data:
https://github.com/CSTR-Edinburgh/merlin/blob/master/egs/build_your_own_voice/s1/scripts/test_nan.sh

**Comment by yzliu90** [comment]
thank you
cal you tell me what's the additional_features?

**Comment by ronanki** [comment]
additional features can be anything apart from linguistic features extracted from text. 

For example, word embeddings, or bottleneck features etc.,  when you have such features prepared in "data/wemb/*.wemb" and "data/bn/*.bn"

This is how you set them in config file:
additional_features : {'wemb':300, 'bn': 32}

**Comment by syltruong** [comment]
I encountered the same problem. I did check my training files but none of them had a "NaN" value.

I turns out that I had pretraining enabled and that my learning rate was too large (0.004). Lowering it (to 0.002) solved the problem.

Hope this can be useful for others who come across this issue

**Comment by AhensNamar** [comment]
@syltruong May I know where are the pretraining and learning rate variables?

**Comment by ZackHodari** [comment]
Pretraining configuration setting must be set by the user within python https://github.com/CSTR-Edinburgh/merlin/blob/master/src/configuration/configuration.py#L983-L993

Learning rate can be set in a `.conf` file https://github.com/CSTR-Edinburgh/merlin/blob/master/src/configuration/configuration.py#L337
